tirtha_bk/tirtha/views.py

```diff
@@ -163,39 +186,42 @@ def index(request, vid=None, runid=None):
             if not contrib.banned and contrib.active:
                 context.update({"signin_class": ""})
 
+    #block for profile image url
+    try:
+        context["profile_image_url"] = token.get('userinfo').get('picture') #adding the currently signed in profile image to the context
+    except Exception as e:
+        pass
+
     return render(request, template, context)
 
+#-----------------updated code using signin with google
 
 def _signin(token, create=False):
     """
-    Handles `Sign in with Google`. Does the following:
-    * Verifies `token`
-    * Check if contributor exists in DB using details from `token`
-    * If not, create new contributor if `create` is True, else return None
-    * If yes, check if contributor is inactive or banned
-        * If yes, return error
-        * If no, return success
-
+    Handles token-based authentication.
+    Verifies the token and retrieves or creates the contributor.
     """
-    # For development only
+
+    
+
     if not GOOGLE_LOGIN:
         # Return default contributor
         contrib = Contributor.objects.get(email=ADMIN_MAIL)
         output = f"Signed-in as {ADMIN_MAIL}."
         return output, contrib
 
-    contrib = None
     try:
-        # Verify token
-        idinfo = id_token.verify_oauth2_token(
-            token, requests.Request(), GOOGLE_CLIENT_ID
-        )
+        contrib = None
 
         # Get contributor info
-        name = idinfo["name"]
-        email = idinfo["email"]
+        payload = token.get('userinfo')
+        email = payload.get('email')
+        name = payload.get('name')
+  
 
         # Get or create contributor
+        contributor, created = Contributor.objects.get_or_create(email=email, defaults={'active': False})
```

Comment:
1. Add `name` to this call.
2. The `try` block below can be completely removed. Also, remove the `create` arg for this function (`_signin`), as it is unused. However, keep the comments. Move them to before L223 (this line).